skimage/restoration/_denoise.py

```diff
@@ -157,7 +205,18 @@ def denoise_tv_bregman(image, weight, max_iter=100, eps=1e-3, isotropic=True):
     .. [4] https://web.math.ucsb.edu/~cgarcia/UGProjects/BregmanAlgorithms_JacquelineBush.pdf
 
     """
-    return _denoise_tv_bregman(image, weight, max_iter, eps, isotropic)
+    image = np.atleast_3d(img_as_float(image))
+    image = np.ascontiguousarray(image)
+
+    rows = image.shape[0]
+    cols = image.shape[1]
+    dims = image.shape[2]
+
+    shape_ext = (rows + 2, cols + 2, dims)
+
+    out = np.zeros(shape_ext, image.dtype)
+
+    return _denoise_tv_bregman(image, weight, max_iter, eps, isotropic, out)
```

Comment:
I think good design dictates that a function shouldn't mutate a variable AND return a variable.

I know this is a common thread in numpy function design, but I think we should see that as an exception rather than the norm.

Reply:
You can do the `sqeezing` back in python. It also makes more sense to make ti there since `python` created the array. I think `squeezing` returns a view anyway.